ouroboros-network/src/Ouroboros/Network/Protocol/KeepAlive/Type.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE TypeFamilies      #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE EmptyCase         #-}
+{-# LANGUAGE PolyKinds         #-}
+
+-- | The type of the keep alive protocol.
+--
+-- The keep alive protocol is used for
+--
+-- * sending keep alive messages
+-- * making round trip measurements
+--
+-- Each side will run its own version of the keep alive protocol.  It should be
+-- configured so that each side keep away its carrier grade nat from shuting
```

Comment:
It is not just carrier grade nat (CGN) that has this issue - perhaps a better comment might be:

It should be configured so that any intermediate state (such as in customer premise equipment or in a carrier grade NAT) is kept alive.  This has to be a per-node configuration element as this is about the properties of that nodes network connectivity,

Reply:
I'll update the comment, thanks.